### DISCUSSION TITLE
Unable to commission Matter Pressure Sensor

### DISCUSSION THREAD

**Discussion by Astrobirder** [SiliconLabs/arduino#61] [General]
Hello,

I have a Sparkfun Thing Plus Matter board with an Adafruit BME280 Temperature, Pressure and Humidity sensor connected via a QWIIC cable.

I'm using the Arduino IDE version 2.3.2 with the Silicon Labs Arduino Core 2.0.0.

I've merged the example code from the blink example, the matter_temp_sensor (for CPU temperature), and the Adafruit BME280 example.

I'm able to read the temperature, pressure and humidity and have it print on the Serial Monitor without issue, sand Temperature and Humidity both commission, so I'm pretty sure the issue is on the matter side with respect to calls to the pressure sensor piece.

The interesting issue is when I try and commission the device with my Apple TV, 4th generation (yes, with Ethernet & WiFi and the Thread Networking support).

I ONLY have this one device in "My Home".  I've successfully deleted and recommissioned this device dozens of times, by deleting the Bridge/Accessories in the Apple Home app and reburning the bootloader onto the Sparkfun board (which de-commissions the device, by erasing the NVM).   I'm new to all Apple Home and to Matter/Thread, but not new to coding, so take that into consideration.

When the device gets added, the first thing to show up is the new Bridge Device (as all Matter devices are Bridges), and then the Humidity Sensor, and then the Temperature Sensor.  The Pressure sensor is never presented to be registered into the matter network--even though the code for all three sensors is basically copy and paste!  Here are the appropriate pieces of my Arduino Code:

Initial includes/variable definitions:
```
// For Matter:
#include <Matter.h>
#include <MatterTemperature.h>
#include <MatterPressure.h>
#include <MatterHumidity.h>
.
.
MatterTemperature matter_temp_sensor;
MatterPressure matter_pres_sensor;
MatterHumidity matter_humid_sensor;
```

inside of `void setup() {`

```
.
.
.
// For Matter
  Matter.begin();
  delay(200);
  matter_temp_sensor.begin();
  delay(200);
  matter_pres_sensor.begin();
  delay(200);
  matter_humid_sensor.begin();
  
  if (!Matter.isDeviceCommissioned()) {
    Serial.println("Matter device is not commissioned");
    Serial.println("Commission it to your Matter hub with the manual pairing code or QR code");
    Serial.printf("Manual pairing code: %s\n", Matter.getManualPairingCode().c_str());
    Serial.printf("QR code URL: %s\n", Matter.getOnboardingQRCodeUrl().c_str());
  }
  while (!Matter.isDeviceCommissioned()) {
    delay(200);
  }  
  Serial.println("Waiting for Thread network...");
  while (!Matter.isDeviceThreadConnected()) {
    delay(200);
  }
  Serial.println("Connected to Thread network");

  Serial.println("Waiting for Matter device discovery...");
  while (!matter_temp_sensor.is_online()) {
    delay(200);
  }
  while(!matter_pres_sensor.is_online()) {
    delay(200);
  }

  while(!matter_humid_sensor.is_online()) {
    delay(200);
  }

  Serial.println("Matter device is now fully online");
  }
```
  
Inside of my `void loop() {`  I have a call to printValues() (code from the Adafruit Example, which reads the sensor, and publishes to matter:
```

void printValues() {
    Serial.print("Temperature = ");
    Serial.print(bme.readTemperature());
    Serial.println(" °C");

    tempInF = (((9* bme.readTemperature())/5) + 32);
    Serial.print("Temperature = ");
    Serial.print(tempInF);
    Serial.println(" °F");
	
	matter_temp_sensor.set_measured_value_celsius(bme.readTemperature());

    Serial.print("Pressure = ");
    Serial.print(bme.readPressure() / 100.0F);
    Serial.println(" hPa");

    Serial.print("Approx. Altitude = ");
    Serial.print(bme.readAltitude(SEALEVELPRESSURE_HPA));
    Serial.println(" m");

  matter_pres_sensor.set_measured_value(bme.readPressure());

    Serial.print("Humidity = ");
    Serial.print(bme.readHumidity());
    Serial.println(" %");
	
	matter_humid_sensor.set_measured_value(bme.readHumidity());

    Serial.println();
}

```
When commissioning is completed, I see the Temperature and the Humidity on my Apple Home dashboard, but never the Pressure.
I get no errors when compiling or programming the Sparkfun.  I also don't see any errors on the serial console.
Here's a sample of the console output, after including the early commissioning and a few of the value printout loops:
`ON
OFF

Temperature = 24.57 °C
Temperature = 76 °F
Pressure = 993.17 hPa
Approx. Altitude = 168.56 m
Humidity = 65.21 %

ON
OFF`

Any ideas where I'm going wrong?  Are there any known bugs with the Matter Pressure Sensor portion of the library?
I also find it rather odd that I get the Bridge, then Humidity and then the Temperature sensor in that order, as the code has everything in the order of Temperature, Pressure, Humidity, both from a creation of the matter sensor and from reading standpoint.  It's like things are going backwards, but skipping Pressure.

Thank you in advance for your ideas.

**Comment by Astrobirder** [comment]
Here's all of the code, should you be so inclined:
[matter_BME280_sensor.ino.txt](https://github.com/user-attachments/files/15849127/matter_BME280_sensor.ino.txt)

**Comment by silabs-bozont** [comment]
Hello @Astrobirder,

Thank you for the detailed description! I've also noticed the Pressure measurement not working with the Apple ecosystem. It works with the Raspberry Pi and Google hubs.
Your code seems fine. I couldn't find any info confirming whether Apple supports pressure measurement devices. I'll definitely take a look into this and resolve the error if it's on our side!

**Reply by Astrobirder** [reply]
I think you're right. Being a matter noobie, is there a return code that HomeKit sends that would tell me there isn't any pressure sensor support? It isn't clear to me which call into the matter stack causes the registration.
